src/diagnostics.rs

```diff
@@ -23,7 +23,7 @@ pub struct Info {
   err_counter: usize,
   errs: Vec<Error>,
```

Comment:
When I read this it's a bit confusing if errs_with_def copies errs or not.
I don't know what better name we could give these 2 fields, but I think it could be improved

Reply:
maybe program_errors & rule_errors?